front/projectsexport.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use GlpiPlugin\Gantt\ProjectsExport;
+
+include('../../../inc/includes.php');
+
+$project_id = $_GET['project_id'] ?? -1;
+
+// Check right
+Session::checkRight(Project::$rightname, Project::READMY);
+
```

Comment:
I guess we should also call `$project->canViewItem()` to make sure an user can't export someone else project that they are not allowed to see.

Reply:
The `getItemsForProject` method already performs the necessary rights checks.

https://github.com/pluginsGLPI/gantt/blob/36978f33ad541b831bd081a34a1fbe8648e5ca7b/src/ProjectsExport.php#L49-L50

https://github.com/pluginsGLPI/gantt/blob/5701bb66f72d49241f6631d11a9909e1192493cc/src/DataFactory.php#L47-L71